lib/puma/events.rb

```diff
@@ -30,9 +30,6 @@ def initialize(stdout, stderr)
       @stdout = stdout
       @stderr = stderr
 
-      @stdout.sync = true
-      @stderr.sync = true
```

Comment:
I think this still needs to happen otherwise we will have a change in behaviour? These aren't the global STDOUT/STDERR we want to avoid mutating

Reply:
I'm trying to simulate the sync behavior using `flush` what do you think? See https://github.com/puma/puma/blob/89fc1e3ed6f950e65baf4463b41f94871bb3ce5b/lib/puma/events.rb#L67

Reply:
Will take a look later to do it only where needed, but if `flush` does the trick I would prefer it over using `sync=true`

Reply:
Okay, I understand your approach, hehe, but it is confusing. From https://ruby-doc.org/core-2.7.1/IO.html#sync-method

>When sync mode is true, all output is immediately flushed to the underlying operating system and is not buffered by Ruby internally.

I think we need to respect this logic :)

Reply:
We want to set sync to true, but we don't want to do it on `STDOUT`/`STDERR` because people might use these constants to initialise their loggers, where they want a different behaviour.

Reply:
Oh! I was assuming that `flush` and `sync=true` had the same logic and that they both work only at Ruby level. What about `fsync` or `fdatasync` then?  

https://ruby-doc.org/core-2.7.2/IO.html#method-i-fdatasync
https://ruby-doc.org/core-2.7.2/IO.html#method-i-fsync 

Both seem to also work at the operating system level

Do you think that exploring this option worths the effort?

Reply:
Not sure we are dealing with files that needs to be written to disk here?

Reply:
I'll research a bit further but at this point, I think that flushing after every message is just the same as using `sync=true`. If I confirm it would that be enough to move forward?

Reply:
I'm ok with that

Reply:
Hmm, I was confused yesterday, sorry, I get your approach. Should hopefully work.